packages/shell-chrome/src/backend.js

```diff
@@ -113,10 +115,18 @@ function serializeDataProperty(value) {
     }
 }
 
-function discoverComponents(isThroughMutation = false) {
+let components = []
+
+function discoverComponents() {
     var rootEls = document.querySelectorAll('[x-data]')
 
-    var components = []
+    // Exit early if no components have been added or removed
+    const allComponentsInitialized = Object.values(rootEls).every((e) => typeof e.__alpineDevtool !== 'undefined')
```

Comment:
This can be simpler I think
```suggestion
    const allComponentsInitialized = Object.values(rootEls).every((e) => e.__alpineDevtool)
```

Reply:
agreed thanks